src/pages/index.js

```diff
@@ -0,0 +1,355 @@
+const deleteForm = document.forms["delete-form"];
+const deleteModalCloseBtn = deleteModal.querySelector(".modal__close-btn");
+
+const pencileImage = profileCol.querySelector("#pencile-image");
+pencileImage.src = pencile;
+
+const plusImage = profile.querySelector("#plus-image");
+plusImage.src = plus;
+
+const closeImage = cardModalCloseBtn.querySelector("#close-image");
+closeImage.src = close;
+
+const modals = document.querySelectorAll(".modal");
+modals.forEach((modal) => {
+  modal.addEventListener("click", (evt) => {
+    if (evt.target === evt.currentTarget) {
+      closeModal(modal);
+    }
+  });
+});
+
+//store the defualt container to varibales before adding cards
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+function getCardElement(data) {
+  //copy the general content of the template
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImage = cardElement.querySelector("#card-image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+
+  cardNameEl.textContent = data.name;
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  //=====================open image preview form===============
+  cardImage.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewCaptionEl.textContent = data.name;
+  });
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleRemoveCard(cardElement, data._id)
+  );
+
+  cardLikeBtn.addEventListener("click", (evt) =>
+    handleLikeButton(evt, data._id)
+  );
+
+  return cardElement;
+}
+
+//=======================modal functions=============================
+function closeModalOnEscape(evt) {
+  if (evt.key === "Escape") {
+    modals.forEach(closeModal);
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalOnEscape);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.activeElement.blur();
+  document.removeEventListener("keydown", closeModalOnEscape);
+}
+
+//======================= handlers =================================== //TODO set button text for all handlers
+// The form submission handler. Note that its name
+// starts with a verb and concisely describes what it does.
+//global variable to store the submit button
+let submitButton = null
+function handleProfileFormSubmit(evt) {
+  // Prevent default browser behavior, see explanation below.
+  evt.preventDefault();
+
+  //rest API call to update user info
+  //change text content to "saving...""
+  submitButton = evt.submitter;
```

Comment:
Please, don't reassign the submit button **globally**, because your forms may be not inside the modals (just 2 forms on the page),  and it will not be nice to change the submit button of the both forms at the same time


Please, define it with `const` everywhere inside the submit handlers


```js

  const submitButton = evt.submitter;

```